lib/backend/model/bgppeer_test.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2020 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model_test
+
+import (
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	. "github.com/projectcalico/libcalico-go/lib/backend/model"
+	cnet "github.com/projectcalico/libcalico-go/lib/net"
+	"net"
+)
+
+var _ = DescribeTable(
+	"BGP Peer key parsing",
+	func(strKey string, expected Key, isGlobalPeer bool) {
+		var key Key
+		if isGlobalPeer {
+			key = (GlobalBGPPeerListOptions{}).KeyFromDefaultPath(strKey)
+		} else {
+			key = (NodeBGPPeerListOptions{}).KeyFromDefaultPath(strKey)
+		}
+
+		Expect(key).To(Equal(expected))
+		serialized, err := KeyToDefaultPath(expected)
```

Comment:
These tests check for path to key conversion (`KeyFromDefaultPath`), I dint add tests for the other direction (key to path), as we are already making use of `KeyToDefaultPath` for validation.

Reply:
Good use of table driven test